### DISCUSSION TITLE
How to customize a new objective function and run it successfully

### DISCUSSION THREAD

**Discussion by wgtfly** [mdolab/dafoam#712] [Tutorials]
Hello, I know that in DAfoam can customize the objective function, right? I want to know if I want to customize an objective function, such as optimizing the mean square error of the flow field calculation and measurement in a certain area to minimize it, in addition to writing the objective function, do I need to modify your adjoint and primal solvers accordingly? Or can I just redefine the objfunc directly without modifying the solver?

**Comment by friedenhe** [comment]
To add a new objFunc, you need to add a child class to DAObjFunc.C. No need to change the adjoint part. Also, this has been implemented in the DAObjFuncVariance: `optimizing the mean square error of the flow field calculation and measurement in a certain area to minimize it`

**Reply by wgtfly** [reply]
Thanks a lot for your quick response, now I find that S809 tutorial and one of the paper named 'An open-source adjoint-based field inversion tool for data-driven RANS modelling', I would like to know whether it is feasible to change the design variable from beita to something else, such as the classic control point？

**Comment by friedenhe** [comment]
Yes. You can use other design variables, but what is the classic control point? FFD points?
